File: JavaRepo/lab2/src/model/Statement/IStmt.java
package model.Statement;

import exceptions.DivideByZeroException;
import exceptions.InvalidOperandException;
import exceptions.NotExistingException;
import utils.PrgState;

public interface IStmt {
    public String toString();
    public PrgState execute(PrgState state) throws DivideByZeroException, InvalidOperandException, NotExistingException;
}

File: Year 1/MAP/C# - Toy language/Lab10/Toy Language/Toy Language/model/statement/IStmt.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Toy_Language.utils;

namespace Toy_Language.model.statement
{
    public interface IStmt
    {
        PrgState Execute(PrgState state);
    }
}
